src/main/java/mel/tasks/TaskList.java

```diff
@@ -0,0 +1,173 @@
+package mel.tasks;
+
+import mel.exceptions.MelException;
+import mel.exceptions.TaskException;
+import mel.main.Mel;
+import mel.utils.Storage;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * TaskList class that handles a
+ * list of tasks from user input.
+ */
+public class TaskList {
+    private final Mel mel;
+    private final Storage storage;
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Constructs a new TaskList to handle tasks.
+     * @param mel Mel chatbot instance.
+     * @param storage Storage instance handling save file.
+     */
+    public TaskList(Mel mel, Storage storage) {
+        this.mel = mel;
+        this.storage = storage;
+    }
+
+    /**
+     * Handles task input to Mel chatbot.
+     * @param input task input string.
+     * @throws TaskException if input is of invalid format.
+     * @throws TaskException if task input is invalid.
+     */
+    public void taskAction(String input) throws MelException, TaskException {
+        String cmd = input.split(" ", 2)[0];
+        if (Objects.equals(input, "list")) {
+            printAll();
+        } else if (cmd.equals("mark")) {
+            mark(input);
+        } else if (cmd.equals("delete")) {
+            delete(input);
+        } else if (cmd.equals("find")) {
+            find(input);
+        } else {
+            mel.println("Mel remembers...");
+            add(input);
+        }
+    }
+
+    /**
+     * Creates new tasks from input.
+     * @param str task input string.
+     * @throws TaskException if input is of invalid format.
+     * @throws TaskException if task input is invalid.
+     */
+    private void add(String str) throws MelException, TaskException {
```

Comment:
A clearer variable name here would be good! Something like `createTaskFromInput`?